### DISCUSSION TITLE
Create a DB

### DISCUSSION THREAD

**Discussion by ripudamank2** [rusqlite/rusqlite#1098] [Questions]
How to create a DB with the specific name and then store data.
some thing like that
pub fn new<S>(
    module: Option<S>, 
    instance: Option<i32>, 
    name: Option<S>, 
    class: Option<S>
) -> Result<Self> 

Please guide

**Comment by gwenn** [comment]
* https://docs.rs/rusqlite/0.15.0/rusqlite/struct.Connection.html#method.open
* https://docs.rs/rusqlite/0.15.0/rusqlite/struct.Connection.html#method.execute_batch or https://docs.rs/rusqlite/0.15.0/rusqlite/struct.Connection.html#method.execute

**Comment by gwenn** [comment]
Would you mind closing the issue ?

**Comment by ripudamank2** [comment]
I had one question before closing this issue
error:
 note: expected type `std::option::Option<std::vec::Vec<u8>>`
              found type `std::vec::Vec<kstat::KstatData>`
How i can convert found to expected one please guide

**Comment by gwenn** [comment]
https://docs.rs/kstat/0.1.0/kstat/struct.KstatData.html ?
How does it relate to `rusqlite` ?

**Comment by ripudamank2** [comment]
i want to store kstat data in DB

**Comment by ripudamank2** [comment]
the trait `rusqlite::types::ToSql` is not implemented for `std::collections::HashMap<std::string::String, String>
Would it be possible to implement rusqlite for Hasmap?

**Comment by gwenn** [comment]
I guess you can use [serde](https://crates.io/crates/serde) and store the result/bytes in a blob.

**Comment by ripudamank2** [comment]
Would it be possible to store struct (data) in vec<>
